src/cmd/pkg/auth/auth.go

```diff
@@ -0,0 +1,234 @@
+// Copyright 2024 Chronosphere Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package auth provides utilities for authenticating a chronoctl session.
+package auth
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"time"
+
+	"github.com/pkg/browser"
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+
+	"github.com/chronosphereio/chronoctl-core/src/cmd/pkg/env"
+	"github.com/chronosphereio/chronoctl-core/src/cmd/pkg/groups"
+	"github.com/chronosphereio/chronoctl-core/src/cmd/pkg/output"
+	"github.com/chronosphereio/chronoctl-core/src/cmd/pkg/token"
+)
+
+const (
+	defaultLoginPath = "/login/cli"
+	defaultOrgPath   = "default-org"
+)
+
+var (
+	errMustIncludeOrgName = errors.New("A single organization name is required")
+)
+
+// NewChronoctlStore creates a new token store in the user's local cache directory to store short-lived chronoctl credentials
+func NewChronoctlStore() (*token.Store, error) {
+	cacheDir, err := os.UserCacheDir()
+	if err != nil {
+		return nil, errors.WithStack(err)
+	}
+	chronoctlCacheDir := filepath.Join(cacheDir, "chronoctl")
+	store := token.NewFileStore(chronoctlCacheDir)
+	return store, nil
+}
+
+// NewCommand returns a command for Chronosphere authentication
+func NewCommand() *cobra.Command {
+	root := &cobra.Command{
+		Use:   "auth",
+		Short: "All commands related to authenticating the chronoctl session.",
+	}
+
+	root.AddGroup(groups.Commands)
+	c := subcommand{}
+	root.AddCommand(
+		c.newAuthLoginCmd(browser.OpenURL),
+		c.newSetDefaultOrgCmd(),
+		c.newPrintAccessTokenCmd(),
+		c.newListCmd(),
+	)
+	return root
+}
+
+type subcommand struct {
+	store *token.Store
+}
+
+// getStore returns a token store contained in the subcommand if one is defined, otherwise returns a NewChronoctlStore
+func (c *subcommand) getStore() (*token.Store, error) {
+	if c.store != nil {
+		return c.store, nil
+	}
+	store, err := NewChronoctlStore()
+	if err != nil {
+		return nil, errors.WithStack(err)
+	}
+	return store, nil
+}
+
+type loginOpts struct {
+	orgName           string
+	loginPath         string
+	skipSetDefaultOrg bool
+}
+
+func (o *loginOpts) registerFlags(cmd *cobra.Command) {
```

Comment:
nit: addFlags to match other opts

Reply:
SGTM - [`3975d5c` (#27)](https://github.com/chronosphereio/chronoctl-core/pull/27/commits/3975d5c62f75b1f6382a4c154e63a518d622b7c6)